docs/developers/deprecating.md

```diff
@@ -24,7 +23,7 @@ We will now describe the steps which should be taken to deprecate various types
 The following things need to be done whenever a feature is being declared as deprecated:
 
  1. Update documentation to include alternative, supported way(s) of accomplishing what is now deprecated. If no alternative is available, include a note indicating that use of the deprecated API should be avoided. Updating the documentation includes adding a `\deprecated` tag with the aforementioned notes to the Doxygen for any public declarations associated with the feature.
- 2. Add a "deprecated" attribute specifier (see [https://en.cppreference.com/w/cpp/language/attributes/deprecated](https://en.cppreference.com/w/cpp/language/attributes/deprecated)) to the C/C++ public declarations specific to this API and include a message that is similar if not identical to the message used in (1). This includes any functions, structs, classes, and typedefs. If a struct, class, or typedef is being used by other APIs which are not being deprecated and the usage is not related to the feature which is being deprecated, do not deprecate these. For example, `DataObjInp` could be used by an API which is being deprecated, but `DataObjInp` is not specific to that API. Therefore, `DataObjInp` would not be a candidate for deprecation.
+ 2. Add a "deprecated" attribute specifier (`[[deprecated]]` for C++ or `__attribute__(deprecated)` for C) to the C/C++ public declarations specific to this API and include a message that is similar if not identical to the message used in (1). This includes any functions, structs, classes, and typedefs. If a struct, class, or typedef is being used by other APIs which are not being deprecated and the usage is not related to the feature which is being deprecated, do not deprecate these. For example, `DataObjInp` could be used by an API which is being deprecated, but `DataObjInp` is not specific to that API. Therefore, `DataObjInp` would not be a candidate for deprecation.
```

Comment:
Let's replace `typedef` with `type alias`.

That covers both `using` directives and `typedef`s.

Reply:
Done